src/symbolize/mod.rs

```diff
@@ -117,6 +118,29 @@ pub use symbolizer::Symbolizer;
 use crate::Addr;
 
 
+/// A enumeration of the different input types the symbolization APIs
+/// support.
+#[derive(Clone, Copy, Debug)]
+pub enum Input<A, O> {
+    /// A virtual address.
+    ///
+    /// A virtual address is an address as a process would see it, for
+    /// example. A virtual address may contain relocation or address space
+    /// randomization effects.
+    VirtAddr(A),
+    /// A file address.
+    ///
+    /// A file address is an address as it would appear in a binary or
+    /// debug symbol file, i.e., one excluding any relocations.
+    FileAddr(A),
+    /// A file offset.
+    ///
+    /// A file offset is the linear offset of a byte in a file in the file
```

Comment:
"in a file in the file system from its start" reads a bit awkwardly... "A file offset within the corresponding ELF binary"?